### PR TITLE
@craigspaeth => Add check for empty email_metadata

### PR DESCRIPTION
**Pull request by eessex**
Email metadata form was failing when article has no email_metadata attribute, adds an empty object if this is the case. 

Related convo - https://artsy.slack.com/archives/C04GGGDCM/p1513198964000377

### CODE REVIEW

**Comment by craigspaeth** [comment]
This is a good example of comments I've left in the past about defaulting to "empty" data types instead of allowing nullable fields. In this case, we might want to be more strict at the [API schema level](https://github.com/artsy/positron/blob/master/api/apps/articles/model/schema.coffee#L181). Instead of `.allow(null)` everywhere, maybe we instead do:

````coffeescript
  email_metadata: @object().default({}).keys
    image_url: @string().default('')
    headline: @string().default('')
    author: @string().default('')
    custom_text: @string().default('')
````

around [here](https://github.com/artsy/positron/blob/master/api/apps/articles/model/schema.coffee#L181). In this case, we're modeling that all articles must have email metadata—just some of that data might be not filled in yet.

Let me know if that doesn't work easily/make sense. I think it's worth doing this refactor to the API side though to ensure data integrity in the database.

**Comment by eessex** [comment]
Yes makes sense to me, updated to include default in the schema.

File: client/apps/edit/components/display/components/email.jsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ export class DisplayEmail extends Component {
 
   onChange = (key, value) => {
     const { article, onChange } = this.props
-    const emailMetadata = article.get('email_metadata')
+    const emailMetadata = article.get('email_metadata') || {}
```
Comment (craigspaeth) [inline client/apps/edit/components/display/components/email.jsx]: This part and the test below shouldn't be necessary with the API changes.

File: api/apps/articles/model/schema.coffee [hunk 2]
```diff
@@ -178,7 +178,7 @@ ImageCollectionSection = (->
       id: @string().objectid().allow(null)
       name: @string().allow('', null)
   ]).default([])
-  email_metadata: @object().keys
+  email_metadata: @object().default({}).keys
     image_url: @string().allow('',null)
```
Comment (craigspaeth) [inline api/apps/articles/model/schema.coffee]: Ideally, we'd un-null these too, so...

````coffeescript
    image_url: @string().allow('')
    headline: @string().allow('')
    author: @string().allow('')
    custom_text: @string().allow('')
````

**Comment by craigspaeth** [comment]
Thanks for the update! Left some minor comments, but the important part is in. Feel free to address and merge when ready.

**Pull request merged**